suite-common/wallet-core/src/device/deviceReducer.ts

```diff
@@ -723,3 +723,9 @@ export const selectDeviceLanguage = (state: DeviceRootState) => {
     const features = selectDeviceFeatures(state);
     return features?.language ?? null;
 };
+
+export const selectHasDeviceFirmwareInstalled = (state: DeviceRootState) => {
+    const device = selectDevice(state);
+
+    return device?.firmware !== 'none';
```

Comment:
maybe more robust condition to not return `hasDeviceFirmwareInstalled === true` if there is no device?
```suggestion
    return !!device && device.firmware !== 'none';
```

Reply:
make sense 👍

Reply:
fixed by [ec58dac](https://github.com/trezor/trezor-suite/pull/10635/commits/ec58dac89d70658f4144cb973e486085c6241095)